### PR TITLE
CRM-19221: Disabled role causes php notices

### PR DESCRIPTION
**Pull request by liedekef**
- [CRM-19221:  Disabled role causes php notices](https://issues.civicrm.org/jira/browse/CRM-19221)

### CODE REVIEW

**Comment by monishdeb** [comment]
@liedekef good catch :) Is there any jira ticket? If not can you raise a issue on Jira ?

File: CRM/ACL/Page/ACL.php [hunk 1]
```diff
@@ -185,7 +185,12 @@ public function browse() {
       $acl[$dao->id]['is_active'] = $dao->is_active;
 
       if ($acl[$dao->id]['entity_id']) {
-        $acl[$dao->id]['entity'] = $roles[$acl[$dao->id]['entity_id']];
+        // take disabled roles into account
+        if (isset($roles[$acl[$dao->id]['entity_id']])) {
+           $acl[$dao->id]['entity'] = $roles[$acl[$dao->id]['entity_id']];
+        } else {
+           $acl[$dao->id]['entity'] = "";
+        }
```
Comment (monishdeb) [inline CRM/ACL/Page/ACL.php]: There is util function which does the same job also you can provide default value on empty. Like here 
`$acl[$dao->id]['entity'] = CRM_Utils_Array::value($acl[$dao->id]['entity'], $roles, '');
`

**Comment by liedekef** [comment]
Thanks for the pointerto the utility function. I updated the PR to use the utilty function, and I created the jira CRM-19221 for this.

**Comment by monishdeb** [comment]
Cool :) @liedekef I have another doubt to ask. Is it required to assign empty string '' by default. I mean what if we use CRM_Utils_Array::value($acl[$dao->id]['entity_id'], $roles); instead ?

**Comment by liedekef** [comment]
Well ... according to the class CRM_Utils_Array, the function "value" returns NULL if the array key doesn't exist, which seems to be fine too. So I'll update again :-)

**Comment by monishdeb** [comment]
There is a trick in squashing multiple commits into one. Like in the current PR you got 6 commits and if you want to merge/squash all commits onto 1 then these are the steps : 
1. git checkout yourbranch
2. git rebase -i origin/master 
# Edit command names accordingly
  pick   1fc6c95 My pull request
  squash 6b2481b Hack hack - will be discarded
  squash dd1475d Also discarded

3. git push -f origin/yourbranch 

You can find more details in http://stackoverflow.com/questions/9790448/how-to-update-a-pull-request. Usually its a general practice to keep PR simple with unique commit list so to avoid redundancy. Feel free to reach us for any of your doubts on https://chat.civicrm.org/civicrm/channels/dev :)

**Comment by liedekef** [comment]
I don't know much of git ... I do all these edits via github (I find git to be needlessly complicated compared with svn or old cvs). I've spent the last 2 hours reading howto's and stackoverflows on how to get a remote branch locally and finally came up with what seems to be almost working:

git clone https://github.com/liedekef/civicrm-core.git
cd civicrm-core/
git fetch origin
git branch -v -a
git checkout -b patch-1 remotes/origin/patch-1
git rebase -i origin/master ==> there delete all that don't interest you, keep the first and squash the others

But then the push doesn't work ... so currently I give up (tried pull/rebase/push,more rm -rf, clone again etc ...)
Apparently git isn't made for simple people ...

**Comment by jitendrapurohit** [comment]
Since there is a false commit in the PR, can you rebase your branch please, or create a new PR ?

**Comment by liedekef** [comment]
I'll create a new one :-)

**Pull request closed without merging**